### DISCUSSION TITLE
GNU readline keys

### DISCUSSION THREAD

**Discussion by jbakosi** [daniele77/cli#151] [Q&A]
Is it possible to implement recognizing basic keys from the [GNU readline library](https://tiswww.case.edu/php/chet/readline/readline.html)? Perhaps via optionally depending on it. That would give users who are used to unix shells a comfortable environment when using the cli.

**Comment by daniele77** [comment]
Hi @jbakosi ,
yes, it would be helpful to add some "shortcut" bash-like. I'm going to add this to the following task list.
Do you have a list of the most useful that the library should support?
Thank you.

**Reply by jbakosi** [reply]
I think if only a partial support is implemented from readline, the list will be subjective, which is not necessarily a problem, of course. Are you against adding it as an optional dependency? (Also, I'm not sure how hard that would be.)

In any case, my subjective list of keys are:
>C-a Move to the start of the line. 
> C-e Move to the end of the line. 
> M-f Move forward a word, where a word is composed of letters and digits. 
> M-b Move backward a word. 
> C-l Clear the screen, reprinting the current line at the top.
> C-k Kill the text from the current cursor position to the end of the line.
> C-u Kill the text from the current cursor position to the beginning of the line.
> C-d Exit.

**Comment by jbakosi** [comment]
I think C-d already works.

**Reply by daniele77** [reply]
Well, actually the cli library should remain independent of third-party libraries, by design (it only depends on `asio` but just if you want the telnet server).
Moreover, I think `readline` is not portable, and it's a C library, and really I don't want to replace half of my library with another one :-)
Adding new keystrokes to the current implementation is not a big deal, so I guess it's better I'll try to implement them at hand.

**Comment by jbakosi** [comment]
Yeah, I completely understand. A couple of new keystrokes will probably be easier to implement.

**Comment by daniele77** [comment]
Opened new issue (#152)
